### PR TITLE
chore(refactor): Refactor code for custom orders route

### PR DESCRIPTION
**Pull request by prateekj117**
Partial fix for #6606 

#### Checklist

- [X] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [X] My branch is up-to-date with the Upstream `development` branch.
- [ ] The unit tests pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] I have added necessary documentation (if appropriate)

- [X] All the functions created/modified in this PR contain relevant docstrings.

### CODE REVIEW

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 1
           

Complexity increasing per file
==============================
- app/api/custom/tickets.py  6
         

Clones added
============
- app/api/custom/tickets.py  2
         

Clones removed
==============
+ app/api/auth.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/open-event-orga-server/open-event-orga-server/pullRequest?prid=4550026&bid=15306428)

File: app/api/custom/tickets.py [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+        try:
+            order = Order.query.filter_by(identifier=order_identifier).first()
+        except NoResultFound:
+            return NotFoundError({'source': ''}, 'This ticket is not associated with any order').respond()
+        if current_user.can_download_tickets(order):
+            key = UPLOAD_PATHS['pdf']['tickets_all'].format(identifier=order_identifier)
+            file_path = '../generated/tickets/{}/{}/'.format(key, generate_hash(key)) + order_identifier + '.pdf'
+            try:
+                return return_file('ticket', file_path, order_identifier)
+            except FileNotFoundError:
```
Comment (niranjan94) [inline app/api/custom/tickets.py]: Codacy Issue found: [undefined name 'FileNotFoundError'](https://app.codacy.com/app/fossasia/open-event-orga-server/pullRequest?prid=4550026)

**Comment by prateekj117** [comment]
@iamareebjamal Please review.

File: app/__init__.py [hunk 2]
```diff
@@ -146,7 +146,8 @@ def create_app():
         from app.api.users import user_misc_routes
         from app.api.orders import order_misc_routes
         from app.api.role_invites import role_invites_misc_routes
-        from app.api.auth import ticket_blueprint, authorised_blueprint
+        from app.api.custom.tickets import ticket_blueprint
```
Comment (iamareebjamal) [inline app/__init__.py]: Move to last

**Comment by prateekj117** [comment]
@iamareebjamal Done.

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 1
           

Complexity increasing per file
==============================
- app/api/custom/tickets.py  6
         

Clones added
============
- app/api/custom/tickets.py  2
         

Clones removed
==============
+ app/api/auth.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/open-event-orga-server/open-event-orga-server/pullRequest?prid=4550026&bid=15306428)

File: app/api/custom/tickets.py (hunk 1 above)
Comment (niranjan94) [inline app/api/custom/tickets.py]: Codacy Issue found: [undefined name 'FileNotFoundError'](https://app.codacy.com/app/fossasia/open-event-orga-server/pullRequest?prid=4550026)

**Comment by prateekj117** [comment]
Depends on [#6615](https://github.com/fossasia/open-event-server/pull/6615)

**Comment by iamareebjamal** [comment]
How can 6610 depend on 6615?

**Comment by prateekj117** [comment]
@iamareebjamal I made the branch of that PR out of the branch of this PR only.

**Comment by iamareebjamal** [comment]
Exactly, so how can this PR depend on that PR?

**Comment by iamareebjamal** [comment]
A parent births a child, so does the child depend on the parent or parent depends on the child? This is the parent, that is the child

**Comment by prateekj117** [comment]
@iamareebjamal Please merge.

**Comment by iamareebjamal** [comment]
Fix the PR dependency first

**Comment by prateekj117** [comment]
@iamareebjamal Done.

File: app/api/custom/tickets.py [hunk 3]
```diff
@@ -0,0 +1,139 @@
+from flask import Blueprint, jsonify, request
```
Comment (iamareebjamal) [inline app/api/custom/tickets.py]: Rename to orders.py

**Comment by prateekj117** [comment]
@iamareebjamal Done.

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 1
           

Complexity increasing per file
==============================
- app/api/custom/orders.py  6
         

Clones added
============
- app/api/custom/orders.py  2
         

Clones removed
==============
+ app/api/auth.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/open-event-orga-server/open-event-orga-server/pullRequest?prid=4550026&bid=15306428)

File: app/api/custom/orders.py [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+        try:
+            order = Order.query.filter_by(identifier=order_identifier).first()
+        except NoResultFound:
+            return NotFoundError({'source': ''}, 'This ticket is not associated with any order').respond()
+        if current_user.can_download_tickets(order):
+            key = UPLOAD_PATHS['pdf']['tickets_all'].format(identifier=order_identifier)
+            file_path = '../generated/tickets/{}/{}/'.format(key, generate_hash(key)) + order_identifier + '.pdf'
+            try:
+                return return_file('ticket', file_path, order_identifier)
+            except FileNotFoundError:
```
Comment (niranjan94) [inline app/api/custom/orders.py]: Codacy Issue found: [undefined name 'FileNotFoundError'](https://app.codacy.com/app/fossasia/open-event-orga-server/pullRequest?prid=4550026)

**Pull request merged**